ir/metadata/specialized_metadata.go

```diff
@@ -1968,8 +1968,9 @@ type DITemplateTypeParameter struct {
 	// (optional) Distinct.
 	Distinct bool
 
-	Name string // optional; empty if not present.
-	Type Field  // required.
+	Name      string // optional; empty if not present.
+	Type      Field  // required.
+	Defaulted *bool  // optional; nil if not present.
```

Comment:
Similar to PR #147, and for consistency, in https://github.com/llir/llvm/pull/157#discussion_r520637325 we changed from `*int64` to `int64` and relied on the zero value for omitted values.

For the same reason, I'd prefer to keep a consistent API and change from `*bool` to `bool` for the `Defaulted` field.

```suggestion
	Defaulted bool  // optional; zero value if not present.
```

The same goes for `DITemplateValueParameter.Defaulted`

Reply:
Note, it could be the case that setting `Defaulted` to `false` is not the same as omitting Defaulted. In which case, the above mentioned approach may not be sufficient.

Reply:
I can't find a reference about this field, and I would push commit for `LLString`, maybe point out this problem but postpone until we get more information?